tests/framework/pvc_utils.go

```diff
@@ -0,0 +1,192 @@
+	err := wait.PollImmediate(2*time.Second, DefaultTimeout, func() (bool, error) {
+		var err error
+		pod, err = f.K8sClient.CoreV1().Pods(f.Namespace.GetName()).Create(podDef)
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		Fail("Unable to create Pod: " + podName + ", error: " + err.Error())
+	}
+	return pod
+}
+
+// Delete the passed in Pod from the passed in Namespace
+func (f *Framework) DeletePodFromNamespace(pod *k8sv1.Pod, namespace *k8sv1.Namespace) {
+	err := wait.PollImmediate(2*time.Second, DefaultTimeout, func() (bool, error) {
+		err := f.K8sClient.CoreV1().Pods(namespace.GetName()).Delete(pod.GetName(), &metav1.DeleteOptions{})
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		Fail("Unable to delete Pod: " + pod.GetName() + ", error: " + err.Error())
+	}
+}
+
+// Delete the passed in Pod from the Framework primary namespace
+func (f *Framework) DeletePod(pod *k8sv1.Pod) {
+	f.DeletePodFromNamespace(pod, f.Namespace)
+}
+
+func newExecutorPodWithPVC(podName, cmd string, pvc *k8sv1.PersistentVolumeClaim) *k8sv1.Pod {
+	return &k8sv1.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: podName},
+		Spec: k8sv1.PodSpec{
+			RestartPolicy: k8sv1.RestartPolicyNever,
+			Containers: []k8sv1.Container{
+				{
+					Name:    "runner",
+					Image:   "fedora:28",
+					Command: []string{"/bin/sh", "-c", "sleep 5; echo I am an executor pod;"},
+					VolumeMounts: []k8sv1.VolumeMount{
+						{
+							Name:      pvc.GetName(),
+							MountPath: defaultPvcMountPath,
+						},
+					},
+				},
+			},
+			Volumes: []k8sv1.Volume{
+				{
+					Name: pvc.GetName(),
+					VolumeSource: k8sv1.VolumeSource{
+						PersistentVolumeClaim: &k8sv1.PersistentVolumeClaimVolumeSource{
+							ClaimName: pvc.GetName(),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (f *Framework) waitTimeoutForPodReady(podName string, timeout time.Duration) error {
+	return wait.PollImmediate(2*time.Second, timeout, f.podRunning(podName, f.Namespace.GetName()))
+}
+
+func (f *Framework) podRunning(podName, namespace string) wait.ConditionFunc {
+	return func() (bool, error) {
+		pod, err := f.K8sClient.CoreV1().Pods(namespace).Get(podName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		switch pod.Status.Phase {
+		case k8sv1.PodRunning:
+			return true, nil
+		case k8sv1.PodFailed, k8sv1.PodSucceeded:
+			return false, nil
+		}
+		return false, nil
+	}
+}
+
+// Verify a particular PVC is empty, returns true if it is, false otherwise.
+func (f *Framework) VerifyPVCIsEmpty(pvc *k8sv1.PersistentVolumeClaim) bool {
+	executorPod := f.CreateExecutorPodWithPVC("verify-pvc-empty", pvc)
+	err := f.waitTimeoutForPodReady(executorPod.GetName(), DefaultTimeout)
+	Expect(err).To(BeNil())
+	output := f.ExecShellInPod(executorPod.GetName(), "ls -1 /pvc | wc -l")
+	f.DeletePod(executorPod)
+	return strings.Compare("0", output) == 0
```

Comment:
Is there no way to capture the exit code of the ls command?

Reply:
I am not sure I understand what you are getting at, how would the exit code tell me if the PVC is empty or not? Or are you trying to see if something happened with the ls command?

Reply:
It would only tell you if the file is present which it always will be since it's mounted. So nevermind...